migrations/env.py

```diff
@@ -57,7 +57,8 @@ def run_migrations_online():
     connection = engine.connect()
     context.configure(
                 connection=connection,
-                target_metadata=target_metadata
+                target_metadata=target_metadata,
+                render_as_batch=True
```

Comment:
Yeah. It turns out flask-migrate did not "fix itself." I kept running into issues trying to drop columns. I think what you tested for that issue must have been just adding columns, which works fine with sqlite. Render_as_batch writes the migration script as a batch migration. If the migration occurs on a sqlite table, it makes a new table with the modified columns and then copies over the data. If it's on another db, it just processes each step as usual. You can read more about it here: http://alembic.readthedocs.org/en/latest/batch.html